src/CallSettings.php

```diff
@@ -217,30 +191,34 @@ public function getUserHeaders()
     public function merge(CallSettings $otherSettings = null)
     {
         if (is_null($otherSettings)) {
-            return new CallSettings([
-                'timeoutMillis' => $this->timeoutMillis,
-                'retrySettings' => $this->retrySettings,
-                'userHeaders' => $this->userHeaders,
-            ]);
-        } else {
-            $timeoutMillis = $this->timeoutMillis;
-            if ($otherSettings->getTimeoutMillis() != self::INHERIT_TIMEOUT) {
-                $timeoutMillis = $otherSettings->getTimeoutMillis();
-            }
-            $retrySettings = $this->retrySettings;
-            if (is_null($otherSettings->getRetrySettings())
-                || !$otherSettings->getRetrySettings()->shouldInherit()) {
-                $retrySettings = $otherSettings->getRetrySettings();
-            }
-            $userHeaders = $this->userHeaders;
-            if (!is_null($otherSettings->userHeaders)) {
-                $userHeaders = $otherSettings->userHeaders;
-            }
-            return new CallSettings([
-                'timeoutMillis' => $timeoutMillis,
-                'retrySettings' => $retrySettings,
-                'userHeaders' => $userHeaders,
-            ]);
+            return $this->with([]);
+        }
+        return $this->with($otherSettings->toArray());
+    }
+
+    /**
+     * @return array
+     */
+    private function toArray()
+    {
+        $arr = [];
+        if (!is_null($this->getRetrySettings())) {
+            $arr['retrySettings'] = $this->getRetrySettings();
+        }
+        if (!is_null($this->getUserHeaders())) {
+            $arr['userHeaders'] = $this->getUserHeaders();
+        }
+        return $arr;
+    }
+
+    private static function convertArrayFromSnakeCase($settings)
```

Comment:
It seems like a common operation. Shouldn't it be in some utility class instead?

Reply:
I would rather not expose this method publicly if possible, as it is only used here. Moving it to a different class would require it to be public.

If we find a broader use for it, we can move it into a utility class at that point.